editor/filesystem_dock.cpp

```diff
@@ -2134,17 +2134,15 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
 
 #ifdef LINUXBSD_ENABLED
 			String chosen_terminal_emulator;
+			String pipe;
 			for (const String &terminal_emulator : terminal_emulators) {
 				List<String> test_args; // Required for `execute()`, as it doesn't accept `Vector<String>`.
+				test_args.push_back("|");
+				test_args.push_back("command");
 				test_args.push_back("-v");
 				test_args.push_back(terminal_emulator);
-				// Silence command name being printed when found. (stderr is already silenced by `OS::execute()` by default.)
-				// FIXME: This doesn't appear to silence stdout.
-				test_args.push_back(">");
-				test_args.push_back("/dev/null");
-				int exit_code = 0;
-				const Error err = OS::get_singleton()->execute("command", test_args, nullptr, &exit_code);
-				if (err == OK && exit_code == EXIT_SUCCESS) {
+				const Error err = OS::get_singleton()->execute("bash", test_args, &pipe);
+				if (err == OK && pipe.find("/")) {
```

Comment:
```suggestion
				if (err == OK && pipe.contains("/")) {
```
Unless you mean to return true if `pipe` contains either no `/`, or it isn't the first character, if that's desired it should be made clear that this is what's desired

Reply:
For  `bash | command -v gnome-terminal` you get the full path:
/usr/bin/gnome-terminal

and for `bash | command -v gnome-terminalBADCMD` I guess only new line or EOL...

Reply:
Are you trying to check for if there is a `/` in the string?

Reply:
The | does not work for some reason. 
Maybe try calling the terminals with a --help option and see if the call succeeds...

Reply:
The list elements are surrounded by double quotes in `os_unix.cpp`. 
So now it works as intended.